traffic/traffic_info.cpp

```diff
@@ -398,17 +405,17 @@ bool TrafficInfo::ReceiveTrafficKeys()
   return true;
 }
 
-bool TrafficInfo::ReceiveTrafficValues(string & etag, vector<SpeedGroup> & values)
+TrafficInfo::DataState TrafficInfo::ReceiveTrafficValues(string & etag, vector<SpeedGroup> & values)
```

Comment:
Can't you update m_dataState and check it?
Overall, what we do now is VERY dirty: we basically have 3 separate classes for
data availability, two here and one in traffic manager.